agents/base/Agent.java

```diff
@@ -33,15 +33,17 @@
     private LogWrapper logWrapper;
     private ThreadContext context;
 
-    void init(AgentContext agentContext, Random random, T item, String sessionKey, String tracker, Logger logger) {
+    void init(AgentContext agentContext, Random random, T item, String sessionKey, String tracker, Logger logger, Boolean trace) {
```

Comment:
Instead of adding `Boolean trace` as a parameter everywhere, it should be on an additional getter on `agentContext` i.e. `agentContext.shouldTrace()`. `AgentContext` represents all of the "global" parameters for an iteration. We could also clean things up further here by renaming `AgentContext` to something like `SimulationContext` or `IterationContext` and wrapping the other parameters, `random`, `item`, `sessionKey`, `tracker` and `logger` into a proper `AgentContext` class.

Reply:
For now, we can just add `shouldTrace` to `agentContext`

Reply:
Ok great!

Reply:
I agree with the renaming to `IterationContext`, I've gone that far